From b91115633dbf6f64745fda2db7fbb83f918dd500 Mon Sep 17 00:00:00 2001
From: Ted Turocy <ted.turocy@gmail.com>
Date: Fri, 7 Jul 2017 12:56:56 +0100
Subject: [PATCH] Const-correctness fix in shared_ptr.h

This corrects the parameter to weak_ptr::swap(), which had been
incorrectly labeled as const.
---
 src/libgambit/shared_ptr.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libgambit/shared_ptr.h b/src/libgambit/shared_ptr.h
index 959510e5..4379840e 100644
--- a/src/libgambit/shared_ptr.h
+++ b/src/libgambit/shared_ptr.h
@@ -131,7 +131,7 @@ template <class T> class weak_ptr {
   long use_count(void) const { return *m_count; }
   bool expired(void) const   { return *m_count == 0; }
 
-  void swap(const weak_ptr<T> &other)  // never throws
+  void swap(weak_ptr<T> &other)  // never throws
   {
     std::swap(m_ptr, other.m_ptr);
     std::swap(m_count, other.m_count);
